video/csputils.h

```diff
@@ -139,6 +139,24 @@ extern const struct m_opt_choice_alternatives mp_stereo3d_names[];
 #define MP_STEREO3D_NAME_DEF(x, def) \
     (MP_STEREO3D_NAME(x) ? MP_STEREO3D_NAME(x) : (def))
 
+struct mp_hdr_metadata {
+    // HDR10
+    // Mastering display metadata
+    float min_luma, max_luma;       // min/max luminance (in cd/m²)
+
+    // Content light level
+    float max_cll;                  // max content light level (in cd/m²)
+    float max_fall;                 // max frame average light level (in cd/m²)
+
+    // HDR10+
+    float scene_max[3];             // maxRGB in cd/m² per component (RGB)
+    float scene_avg;                // average of maxRGB in cd/m²
+
+    // CIE Y
+    float max_pq_y;                 // maximum PQ luminance in cd/m²
+    float avg_pq_y;                 // averaged PQ luminance in cd/m²
```

Comment:
Because it is convenient down the line. I don't expect mpv to process this data in any meaningful way anyway.

I can keep it in PQ, but then I need to convert in in lua probably, because nits display is more understandable for users.

Reply:
Food for thought: In plplay we display *both* PQ% *and* convert to nits for display.

I'd rather we keep `mp_hdr_metadata` in PQ simply because now that this struct exists we should probably deprecate `sig_peak` etc. and start using `mp_hdr_metadata` internally, including mapping it directly from AVFrame.

Reply:
![image](https://github.com/mpv-player/mpv/assets/1149047/47deed29-2e2c-4b54-8af3-07759ec9c40e)

Reply:
Ok, fair, let me refactor this part.